macro-utils/src/tuple.rs

```diff
@@ -23,12 +23,12 @@ fn from_str(ast: &syn::DeriveInput) -> syn::Result<TokenStream> {
     let convert = match extract_field_type(ast) {
         Type::String(_) => quote! { s.to_string() },
         Type::Struct(ty) | Type::Simple(ty) => {
-            quote! { #ty::from_str(s).map_err(|e| e.to_string())? }
+            quote! { <#ty as ::std::str::FromStr>::from_str(s).map_err(|e| e.to_string())? }
```

Comment:
Why is it needed to explicitly cast to FromStr?

Reply:
It's not strictly needed, but it's good practice to invoke trait methods this way in generated code. Without this, if a type had both a `FromStr` implementation and an inherent `from_str` method, the latter would be selected.